src/control/dmg/pool.go

```diff
@@ -23,7 +23,177 @@
+	NVMeSize   string `short:"n" long:"nvme-size" description:"Size of NVMe component of DAOS pool"`
+	RankList   string `short:"r" long:"ranks" description:"DAOS pool to be owned by given user"`
+	NumSvcReps uint32 `short:"v" long:"nsvc" default:"1" description:"Number of pool service replicas"`
+	//GID uint32 `short:"g" long:"gid" description:"Pool GID"`
+	//UID uint32 `short:"u" long:"uid" description:"Pool UID"`
+	//Group string `short:"G" long:"group" description:"Pool group name"`
+	//Mode uint64 `short:"m" long:"mode" description:"Pool permissions mode"`
+}
+
+// getSize retrieves number of bytes from human readable string representation
+func getSize(sizeStr string) (bytesize.ByteSize, error) {
+	if sizeStr == "" {
+		return bytesize.New(0.00), nil
+	}
+	if common.IsAlphabetic(sizeStr) {
+		return bytesize.New(0.00), errors.New(msgSizeNoNumber)
+	}
+
+	// change any alphabetic characters to upper before ByteSize.parse()
+	sizeStr = strings.ToUpper(sizeStr)
+
+	// append "B" character if absent (required by ByteSize.parse())
+	if !strings.HasSuffix(sizeStr, "B") {
+		sizeStr += "B"
+	}
+
+	return bytesize.Parse(sizeStr)
+}
+
+// calcStorage calculates SCM & NVMe size for pool from user supplied parameters
+func calcStorage(scmSize string, nvmeSize string) (
+	scmBytes bytesize.ByteSize, nvmeBytes bytesize.ByteSize, err error) {
+
+	scmBytes, err = getSize(scmSize)
+	if err != nil {
+		err = errors.WithMessagef(
+			err, "illegal scm size: %s", scmSize)
+		return
+	}
+
+	if scmBytes == 0 {
+		err = errors.New(msgSizeZeroScm)
+		return
+	}
+
+	nvmeBytes, err = getSize(nvmeSize)
+	if err != nil {
+		err = errors.WithMessagef(
+			err, "illegal nvme size: %s", nvmeSize)
+		return
+	}
+
+	ratio := 1.00
+	if nvmeBytes > 0 {
+		ratio = float64(scmBytes) / float64(nvmeBytes)
+	}
+
+	if ratio < 0.01 {
+		fmt.Printf(
+			"SCM:NVMe ratio is less than 1%%, DAOS performance " +
+				"will suffer!\n")
+	}
+	fmt.Printf(
+		"Creating DAOS pool with %s SCM and %s NvMe storage "+
+			"(%.3f ratio)\n",
+		scmBytes.Format("%.0f", "", false),
+		nvmeBytes.Format("%.0f", "", false),
+		ratio)
+
+	return scmBytes, nvmeBytes, nil
+}
+
+func parseRanks(ranksStr string) (ranks []uint32, err error) {
+	for i, rankStr := range strings.Split(ranksStr, ",") {
+		rank, err := strconv.ParseUint(rankStr, 10, 32)
+		if err != nil {
+			return ranks, errors.WithMessagef(err, "element %d", i)
+		}
+
+		ranks = append(ranks, uint32(rank))
+	}
+
+	return
+}
+
+// createPool with specified parameters on all connected servers
+func createPool(
+	groupName string, userName string, aclFile string, scmSize string,
+	nvmeSize string, rankList string, numSvcReps uint32) error {
+
+	scmBytes, nvmeBytes, err := calcStorage(scmSize, nvmeSize)
+	if err != nil {
+		return errors.Wrap(err, "calculating pool storage sizes")
+	}
+
+	// TODO: do we want to verify format of list here or just at iosrv?
```

Comment:
perhaps a related issue is that if the user does not provide a rank list at all I think the behavior of the current dmg is to use all of the ranks of the servers in the process group. Looking at existing dmg client side implementation quickly I see an attach to the pool server process group name inside dc_pool_create(); and in this scenario I do not see a resolution of the rank list on the client side - so am assuming then it gets resolved on the server side.

Reply:
yes I will copy that approach, I think we answered this TODO when talking, we verify we have a list of positive integers separated by commas, and send list of uint32s in proto message, empty list recognised as special case iosrv side

Reply:
noted